dora/src/object.rs

```
@@ -517,7 +521,7 @@ impl Str {
 
             handle
         } else {
-            Ref::null()
```

I don't think we should silently swallow this, better to throw an exception here or make the return type `Option[String]`.

---

This is handled in https://github.com/dinfuehr/dora/pull/152/files#diff-edf3ee5de6272371b58c0a2d02e7c112.

---

Right sorry, haven't seen that... Couldn't we throw the exception directly from the native code instead? Then we wouldn't need the wrappers (fromStringPart and fromBytesPart) anymore and fromStringPartOrNull and fromBytesPartOrNull can simply replace them.

---

Ah, sorry, I just migrated to native excpetions + options before reading this ... I'll push and then you can decide if it's ok or needs to be reworked.

---

Does this look ok? https://github.com/dinfuehr/dora/pull/152/commits/77e8b3c5fd69f4f9fbdf87565d48b1dc65b0ddae

---

Sure, even better ;)

---

Ah, great! Thanks!